### PR TITLE
IN-124-Daemonset-for-filebeat

### PR DESCRIPTION
**Pull request by fadi-khatib**
# What changed
Create a DaemonSet for file beat that can applied to the Kubernetes environment:
support 2 configuration options:
1) standard configuration.
2) autodiscovery
  

----

## Pages to review
https://deploy-preview-720--logz-docs.netlify.app/shipping/log-sources/filebeat-kubernetes.html

- LinkToPage1
- LinkToPage2

## Remaining work

- [ ] Technical review
- [ ] Copy Review
- [ ] Redirects: 

## Post launch

To be completed by the docs team upon merge:

- [ ] Teams to update with the new information:
- [ ] Replace original content on the support portal with 'this page has been moved to ...' - paste the URL
- [ ] Update these log shipping pages in the app: - paste the URL

### CODE REVIEW

File: _source/logzio_collections/_log-sources/filebeat-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ kubectl apply -f https://raw.githubusercontent.com/logzio/logz-docs/master/shipping-config-samples/k8s-filebeat.yaml -f https://raw.githubusercontent.com/logzio/logz-docs/master/shipping-config-samples/filebeat-autodiscovery-configuration.yaml
+```
+
+##### If you want to apply  your one custom configuration:
+Download standard-configmap.yaml and apply your changes there, make sure that you will have the same structure of the file.
+  
+```shell
+kubectl apply -f /Users/fadikhatib/logzio/integration-team/logz-docs/_source/logzio_collections/_log-sources/standard-filebeat-daemonset.yaml -f <<Your-custom-configuration-file.yaml>>
+```
+
```
Comment (mirii1994) [inline _source/logzio_collections/_log-sources/filebeat-kubernetes.md:62]: Maybe it's worth to add the download command, something like this - 
```shell
wget link-from-github.com
```
Also, maybe it's worth displaying an example of how a configuration may look? or more extensive note about the _structure_ of the configuration file.
Plus - you accidentally wrote your local path :)

File: shipping-config-samples/k8s-filebeat.yaml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+              name: logzio-logs-secret
+              key: cluster-name
+        - name: IGNORE_OLDER
+          value: 3h
+        - name: LOGZIO_CODEC
+          value: json
+        - name: LOGZIO_TYPE
+          value: filebeat
+        - name: FIELDS_UNDER_ROOT
+          value: "true"
```
Comment (mirii1994) [inline shipping-config-samples/k8s-filebeat.yaml:143]: We'll need to change that. It came to my attention that fields_under_root should always be true, so we need to delete this from the environment variables and to set that field in the configuration to always be true,

```suggestion
```

File: shipping-config-samples/filebeat-standard-configuration.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                logs_path: "/var/log/containers/"
+
+    processors:
+      - add_cloud_metadata: ~
+    fields:
+      logzio_codec: ${LOGZIO_CODEC}
+      token: ${LOGZIO_LOGS_SHIPPING_TOKEN}
+      cluster: ${CLUSTER_NAME}
+      type: ${LOGZIO_TYPE}
+    fields_under_root: ${FIELDS_UNDER_ROOT}
```
Comment (mirii1994) [inline shipping-config-samples/filebeat-standard-configuration.yaml:29]:
```suggestion
    fields_under_root: true
```
Please test that it's working when we're changing it to this :)

**Review by mirii1994 — CHANGES_REQUESTED** [review]
Great!
A few notes -
* When i'm looking at the netlify preview of this build, there are 2 k8s logos, we need to change it to something else. Here's a screen shot:
https://user-images.githubusercontent.com/37478344/97880664-1bf7f900-1d2a-11eb-8279-271cc7d8324c.png
* See my other comments :)

**Comment by mirii1994** [comment]
@fadi-khatib please go over the docs with @shalper , they need some formatting before i'll go over them one last time :)

**Review by fadi-khatib — COMMENTED** [review]
@shalper @mirii1994, one last change.
we don't want to restrict our customers, I have removed the constraints on 'fields'.and added a note regarding the token.

File: _source/logzio_collections/_log-sources/filebeat-kubernetes.md [hunk 4]
```diff
@@ -72,7 +72,10 @@ Run the following command to download the file:
 wget https://raw.githubusercontent.com/logzio/logz-docs/master/shipping-config-samples/filebeat-standard-configuration.yaml
 ```
 
-Apply your custom configuration to the paramaters under `filebeat.yml` and only there. The filebeat.yml field contains a basic Filebeat configuration. You should not change the 'fields' and 'output' fields (indicated in the example below). See Elastic documentation to [learn more about Filebeat configuration options](https://www.elastic.co/guide/en/beats/filebeat/current/configuring-howto-filebeat.html).
```
Comment (fadi-khatib) [inline _source/logzio_collections/_log-sources/filebeat-kubernetes.md]: @shalper @mirii1994, one last change.
we don't want to restrict our customers, I have removed the constraints on 'fields'.and added a note regarding the token.

**Review by fadi-khatib — COMMENTED** [review]
@shalper @mirii1994, one last change.
we don't want to restrict our customers, I have removed the constraints on 'fields'.and added a note regarding the token.

**Review by mirii1994 — APPROVED** [review]
Great work! 👏👏👏

**Pull request merged**